ci/build.py

```diff
@@ -84,31 +85,46 @@ def is_mxnet_root(path: str) -> bool:
         curpath = parent
     return curpath
 
+def buildir() -> str:
+    return os.path.join(get_mxnet_root(), "build")
 
-def container_run(platform: str, docker_binary: str, command: List[str]) -> None:
+def container_run(platform: str, docker_binary: str, command: List[str], dry_run: bool = False) -> str:
     tag = get_docker_tag(platform)
     mx_root = get_mxnet_root()
-    local_build_folder = '{}/build'.format(mx_root)
+    local_build_folder = buildir()
     # We need to create it first, otherwise it will be created by the docker daemon with root only permissions
     os.makedirs(local_build_folder, exist_ok=True)
-    logging.info("Running %s in container %s", command, tag)
     runlist = [docker_binary, 'run', '--rm',
         '-v', "{}:/work/mxnet".format(mx_root), # mount mxnet root
         '-v', "{}:/work/build".format(local_build_folder), # mount mxnet/build for storing build artifacts
         '-u', '{}:{}'.format(os.getuid(), os.getgid()),
         tag]
     runlist.extend(command)
     cmd = ' '.join(runlist)
-    logging.info("Executing: %s", cmd)
-    ret = call(runlist)
-    if ret != 0:
+    if not dry_run:
+        logging.info("Running %s in container %s", command, tag)
+        logging.info("Executing: %s", cmd)
+        ret = call(runlist)
+
+    into_cmd = deepcopy(runlist)
+    idx = into_cmd.index('-u') + 2
+    into_cmd[idx:idx] = ['-ti', '--entrypoint', 'bash']
+    docker_run_cmd = ' '.join(into_cmd)
+
+    if not dry_run and ret != 0:
         logging.error("Running of command in container failed: %s", cmd)
-        into_cmd = deepcopy(runlist)
-        idx = into_cmd.index('-u') + 2
-        into_cmd[idx:idx] = ['-ti', '--entrypoint', 'bash']
-        logging.error("You can try to get into the container by using the following command: %s", ' '.join(into_cmd))
+        logging.error("You can try to get into the container by using the following command: %s", docker_run_cmd)
```

Comment:
Last time I tried copy and pasting this command, I ran into some trouble that the command was not properly runable. I can't give you a specific message right now, but feel free to let me know if you need more details. To reproduce, try to run some commands the CI is executing.